app/code/Magento/Sales/Model/GetCustomerOrdersHistory.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Sales\Model;
+
+use Magento\Framework\Api\Filter;
+use Magento\Framework\Api\Search\FilterGroup;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Api\GetCustomerOrdersHistoryInterface;
+use Magento\Sales\Api\OrderRepositoryInterface;
+
+/**
+ * Class GetCustomerOrdersHistory
+ */
+class GetCustomerOrdersHistory implements GetCustomerOrdersHistoryInterface
+{
+    /**
+     * @var OrderRepositoryInterface
+     */
+    private $orderRepository;
+
+    /**
+     * @param OrderRepositoryInterface $orderRepository
+     */
+    public function __construct(OrderRepositoryInterface $orderRepository)
+    {
+        $this->orderRepository = $orderRepository;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(
+        int $customerId,
+        \Magento\Framework\Api\SearchCriteriaInterface $searchCriteria
+    ) : \Magento\Sales\Api\Data\OrderSearchResultInterface {
+        $this->filterOrdersByCustomerId($customerId, $searchCriteria);
+
+        return $this->orderRepository->getList($searchCriteria);
+    }
+
+    /**
+     * Filter order list with the customer id
+     *
+     * @param int $customerId
+     * @param \Magento\Framework\Api\SearchCriteriaInterface $searchCriteria
+     *
+     * @return void
+     */
+    private function filterOrdersByCustomerId(
+        int $customerId,
+        \Magento\Framework\Api\SearchCriteriaInterface $searchCriteria
+    ) : void {
+        $filter = new Filter(
+            [
+                Filter::KEY_FIELD => OrderInterface::CUSTOMER_ID,
+                Filter::KEY_CONDITION_TYPE => 'eq',
+                Filter::KEY_VALUE => $customerId,
+            ]
+        );
+
+        $groups = $searchCriteria->getFilterGroups();
+
+        $groups[] = (new FilterGroup())->setFilters([$filter]);
```

Comment:
Please, use the filter group factory instead of object creation via the new operator.

Reply:
Solved by using hte FilterGroupBuilder